### ISSUE TITLE
Copy Objects to minio is unusually slow

### ISSUE THREAD

**Issue by deekoder** [minio/mc#2196]
Copy objects to minio is unusual slow

## Expected Behavior

Faster?

## Current Behavior
```
./mc cp --recursive folder/ local/bucket/ is running around 4.5 MB/s
```
```
dd if=/dev/zero of=bigfile bs=1m count=1024
```
```
./mc cp bigfile minio/test is running 128MB/s
```
## Context

The read speed is around 50-70 MB/s . Files under the bucket are something like `BucketName/MediaID/Quality/MovieName/Files`

## Your Environment

Version: 2017-05-05T01:14:51Z
Supermicro SuperStorage 6028R-E1CR12L
CPU : model name : Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
CORE : 40
Memory : 512 GB
Disks : 12 * 4TB
Ubuntu 16.04

**Comment by deekoder** [comment]
Filing for https://github.com/minio/minio/issues/4580

**Comment by curantes** [comment]
I would also like this to be fixed for the mirror command as well.  I see low performance when running mirror to sync from a minio cluster to a filesystem.

**Comment by vadmeste** [comment]
ETA: Oct 13

**Comment by jonaz** [comment]
Status on this one? Im moving 1TB of data at 1MB/s speed with mc mirror command :(

**Comment by harshavardhana** [comment]
> Status on this one? Im moving 1TB of data at 1MB/s speed with mc mirror command :(
> Will take 9.25 days :(

what your client to server bandwidth? @jonaz

**Comment by harshavardhana** [comment]
> what your client to server bandwidth? @jonaz

Also can you perform `iperf -s` on and `iperf -c` to test the effective bandwidth?

**Comment by jonaz** [comment]
@harshavardhana 
iperf client log
```
------------------------------------------------------------
Client connecting to prod-s3stor01, TCP port 5001
TCP window size: 85.0 KByte (default)
------------------------------------------------------------
[  3] local 10.XX.XX.XX port 60796 connected with 10.XX.XX.XX port 5001
[ ID] Interval       Transfer     Bandwidth
[  3]  0.0-10.0 sec  9.74 GBytes  8.37 Gbits/sec

```

iperf server log
```
------------------------------------------------------------
Server listening on TCP port 5001
TCP window size: 85.3 KByte (default)
------------------------------------------------------------
[  4] local 10.XX.XX.XX port 5001 connected with 10.XX.XX.XX port 60796
[ ID] Interval       Transfer     Bandwidth
[  4]  0.0-10.0 sec  9.74 GBytes  8.36 Gbits/sec

```

**Comment by harshavardhana** [comment]
> @harshavardhana
> iperf client log
> 
> ------------------------------------------------------------
> Client connecting to prod-s3stor01, TCP port 5001
> TCP window size: 85.0 KByte (default)
> ------------------------------------------------------------
> [  3] local 10.XX.XX.XX port 60796 connected with 10.XX.XX.XX port 5001
> [ ID] Interval       Transfer     Bandwidth
> [  3]  0.0-10.0 sec  9.74 GBytes  8.37 Gbits/sec
> 
> iperf server log
> 
> ------------------------------------------------------------
> Server listening on TCP port 5001
> TCP window size: 85.3 KByte (default)
> ------------------------------------------------------------
> [  4] local 10.XX.XX.XX port 5001 connected with 10.XX.XX.XX port 60796
> [ ID] Interval       Transfer     Bandwidth
> [  4]  0.0-10.0 sec  9.74 GBytes  8.36 Gbits/sec

@jonaz are these 1TiB files really small like image files?

**Comment by curantes** [comment]
@harshavardhana yes these files are small mostly images and pdf documents.

**Comment by jonaz** [comment]
@harshavardhana yes as @curantes we work together.

**Comment by vadmeste** [comment]
@jonaz, @curantes, I wrote this PR, https://github.com/minio/mc/pull/2289, which not merged/official yet.. but you can test it. It should accelerate mirroring or copying multiple times. I would be happy to assist you.

**Comment by harshavardhana** [comment]
This is going to be fixed by #2289
